python/lsst/meas/algorithms/normalizedCalibrationFlux.py

```diff
@@ -0,0 +1,295 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["NormalizedCalibrationFluxConfig", "NormalizedCalibrationFluxTask"]
+
+import numpy as np
+
+from lsst.afw.image import ApCorrMap
+from lsst.afw.math import ChebyshevBoundedField
+import lsst.pex.config
+import lsst.pipe.base
+from .measureApCorr import MeasureApCorrTask, MeasureApCorrError
+from .sourceSelector import sourceSelectorRegistry
+
+
+class NormalizedCalibrationFluxConfig(lsst.pex.config.Config):
+    """Configuration parameters for NormalizedCalibrationFluxTask.
+    """
+    measure_ap_corr = lsst.pex.config.ConfigurableField(
+        target=MeasureApCorrTask,
+        doc="Subtask to measure aperture corrections.",
+    )
+    raw_calibflux_name = lsst.pex.config.Field(
+        doc="Name of raw calibration flux to normalize.",
+        dtype=str,
+        default="base_CompensatedTophatFlux_12",
+    )
+    normalized_calibflux_name = lsst.pex.config.Field(
+        doc="Name of normalized calibration flux.",
+        dtype=str,
+        default="base_NormalizedCompensatedTophatFlux",
+    )
+    do_set_calib_slot = lsst.pex.config.Field(
+        doc="Set the calib flux slot to the normalized flux?",
+        dtype=bool,
+        default=True,
+    )
+    do_measure_ap_corr = lsst.pex.config.Field(
+        doc="Measure the aperture correction? (Otherwise, just apply.)",
+        dtype=bool,
+        default=True,
+    )
+    fallback_source_selector = sourceSelectorRegistry.makeField(
+        doc="Selector that is used as a fallback if the full aperture correction "
+            "fails.",
+        default="science",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        self.measure_ap_corr.refFluxName = "base_CircularApertureFlux_12_0"
+
+        # This task is meant to be used early when we focus on PSF stars.
+        selector = self.measure_ap_corr.sourceSelector["science"]
+        selector.doUnresolved = False
+        selector.flags.good = ["calib_psf_used"]
+        selector.flags.bad = []
+        selector.signalToNoise.fluxField = self.raw_calibflux_name + "_instFlux"
+        selector.signalToNoise.errField = self.raw_calibflux_name + "_instFluxErr"
+        # Do median for this.
+        self.measure_ap_corr.fitConfig.orderX = 0
+        self.measure_ap_corr.fitConfig.orderY = 0
+
+        fallback_selector = self.fallback_source_selector["science"]
+        fallback_selector.doFluxLimit = False
+        fallback_selector.doFlags = True
+        fallback_selector.doUnresolved = False
+        fallback_selector.doSignalToNoise = False
+        fallback_selector.doIsolated = False
+        fallback_selector.flags.good = ["calib_psf_used"]
+        fallback_selector.flags.bad = []
```

Comment:
I'm missing some connection to Robert's slide here. I was expecting to see a corresponding field for `R1` and `R2`, for the annulus. Is this implicit in `measure_ap_corr` somehow? 🤔

Reply:
That's in `meas_base` which measures a compensated tophat (or compensated gaussian).  This is a task to _normalize_ a generic calibration flux that could be a compensated tophat or compensated gaussian or something else we haven't thought of yet.

Reply:
🤦🏼 I over prepared for this review then. This all LGTM.